lib/queue.rb

```diff
@@ -2,30 +2,61 @@ class Queue
 
   def initialize
     # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    # @list = Array.new(20)
+    @store = Array.new
+    @front_index = 0
+    @back_index = 0
+    @max_size = 20
   end
 
   def enqueue(element)
```

Comment:
This works, but you have implemented the Queue with a straight array instead of a circular buffer.